test/e2e/cache/create_enable_add_with_private_registry_delete.go

```diff
@@ -0,0 +1,283 @@
+			},
+			Data: map[string][]byte{
+				"config.yml": []byte(upstreamConfigYaml),
+			},
+		}
+		Expect(f.ShootFramework.ShootClient.Client().Create(ctx, configSecret)).To(Succeed())
+
+		// create upstream registry service
+		service := &corev1.Service{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "test-registry",
+				Namespace: metav1.NamespaceSystem,
+			},
+			Spec: corev1.ServiceSpec{
+				Selector: map[string]string{
+					"app": "test-registry",
+				},
+				Ports: []corev1.ServicePort{{
+					Name:       "test-registry",
+					Port:       5000,
+					Protocol:   corev1.ProtocolTCP,
+					TargetPort: intstr.FromString("test-registry"),
+				}},
+				Type: corev1.ServiceTypeClusterIP,
+			},
+		}
+		Expect(f.ShootFramework.ShootClient.Client().Create(ctx, service)).To(Succeed())
+
+		// get service cluster IP
+		Expect(f.ShootFramework.ShootClient.Client().Get(ctx, client.ObjectKeyFromObject(service), service)).To(Succeed())
+		upstreamHostPort := service.Spec.ClusterIP + ":5000"
+
+		// create upstream registry pod
+		testRegistry := &corev1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "test-registry",
+				Namespace: metav1.NamespaceSystem,
+				Labels: map[string]string{
+					"app": "test-registry",
+				},
+			},
+			Spec: corev1.PodSpec{
+				Hostname: "test-registry",
+				Containers: []corev1.Container{
+					{
+						Name:  "registry",
+						Image: "registry:2.8.3",
+						Ports: []corev1.ContainerPort{
+							{
+								ContainerPort: 5000,
+								Name:          "test-registry",
+							},
+						},
+						VolumeMounts: []corev1.VolumeMount{
+							{
+								Name:      "htpasswd-volume",
+								MountPath: "/var/lib/password/htpasswd",
+								SubPath:   "htpasswd",
+							},
+							{
+								Name:      "config-volume",
+								MountPath: "/etc/docker/registry",
+							},
+						},
+					},
+				},
+				Volumes: []corev1.Volume{
+					{
+						Name: "config-volume",
+						VolumeSource: corev1.VolumeSource{
+							Secret: &corev1.SecretVolumeSource{
+								SecretName: configSecret.Name,
+							},
+						},
+					},
+					{
+						Name: "htpasswd-volume",
+						VolumeSource: corev1.VolumeSource{
+							Secret: &corev1.SecretVolumeSource{
+								SecretName: htpasswdSecret.Name,
+							},
+						},
+					},
+				},
+			},
+		}
+		Expect(f.ShootFramework.ShootClient.Client().Create(ctx, testRegistry)).To(Succeed())
+		Expect(framework.WaitUntilPodIsRunning(ctx, f.Logger, "test-registry", metav1.NamespaceSystem, f.ShootFramework.ShootClient)).To(Succeed())
+
+		// push nginx:1.24.0 to the upstream registry
+		nodeList, err := framework.GetAllNodesInWorkerPool(ctx, f.ShootFramework.ShootClient, ptr.To("local"))
+		framework.ExpectNoError(err)
+		rootPodExecutor := framework.NewRootPodExecutor(f.Logger, f.ShootFramework.ShootClient, &nodeList.Items[0].Name, "kube-system")
+
+		pullImage := fmt.Sprintf("ctr content fetch --all-platforms docker.io/library/%s > /dev/null", "nginx:1.24.0")
```

Comment:
Can we restrict the pull and push to amd64 and arm64 as these are the only 2 architectures that Gardener supports currently?